script.js

```diff
@@ -1,54 +1,60 @@
-function Animal(name){
+class Animal {
+  constructor(name) {
     this.name = name;
-}
-
-Animal.prototype.action = function(){
-  document.getElementById(this.actionSoundName).play();
-  
-}
-
-Animal.prototype.putInTheDocument = function(){
+  }
+  action() {
+    document.getElementById(this.actionSoundName).play();
+  }
+  putInTheDocument() {
     var petsTable = document.getElementById("petsTable");
     var petTR = document.createElement("tr");
-  
+
     var petNameTD = document.createElement("td");
     petNameTD.textContent = this.name;
     petTR.appendChild(petNameTD);
-  
+
     var petLegsTD = document.createElement("td");
     petLegsTD.textContent = this.legs;
     petTR.appendChild(petLegsTD);
-  
+
     var petActionTD = document.createElement("td");
     var petActionTDButton = document.createElement("button");
     petActionTDButton.textContent = this.actionText;
     petActionTD.appendChild(petActionTDButton);
     petTR.appendChild(petActionTD);
-    
+
     petActionTDButton.onclick = this.action.bind(this);
-    petsTable.querySelector("tbody").appendChild(petTR)
+    petNameTD.onclick = this.openImage.bind(this);
```

Comment:
Bu şekilde yaptığında tıklama görevini sadece bir kolona(td) vermiş oluyorsun. Örneğin legs kolonundaki yere tıkladığında çalışmayacaktır . Bunun yerine satıra bu görevi verebilirsin.